lib/chef/resource/windows_certificate.rb

```diff
@@ -275,14 +272,11 @@ def resolve_thumbprint(thumbprint)
 
         def verify_cert(thumbprint = new_resource.source)
           store = ::Win32::Certstore.open(new_resource.store_name, store_location: native_cert_location)
-          if new_resource.pfx_password.nil?
-            store.valid?(resolve_thumbprint(thumbprint), store_location: native_cert_location, store_name: new_resource.store_name )
-          else
-            store.valid?(resolve_thumbprint(thumbprint), store_location: native_cert_location, store_name: new_resource.store_name)
-          end
+          result = store.valid?(resolve_thumbprint(thumbprint))
```

Comment:
What does `store.valid?` do when `resolve_thumbprint` returns `false`?

Reply:
Separately, why is a method named `valid?` maybe-returning a string?

Reply:
We check that the format of the thumbprint is correct - 40 hex characters with no spaces. If that fails, we throw an exception now (just added that in). Otherwise, we search for a cert that matches that borked thumbprint and return "not found"